### PR TITLE
Added nightly builds for CI

### PR DESCRIPTION
**Pull request by kogeler**
Signed-off-by: Roman Gavrilov <kogeler@gmail.com>

### Description of the Change

Added nightly builds for CI.

It includes:
- x64linux_compiler_list = ['gcc5','gcc7', 'clang6' , 'clang7']
- mac_compiler_list = ['appleclang']
- win_compiler_list = ['msvc']
- testing = true
- testList = '()'
- coverage = true
- cppcheck = true
- sonar = true
- codestyle = true
- useBTF = true
- sanitize = true
- packagePush=false
- doxygen=false

### Benefits

We can check the quality of the code as much as possible in the off-peak hours

### CODE REVIEW

File: .jenkinsci-new/build.groovy [hunk 1]
```diff
@@ -38,8 +38,8 @@ def sonarScanner(scmVars, environment) {
   withEnv(environment) {
     withCredentials([string(credentialsId: 'SONAR_TOKEN', variable: 'SONAR_TOKEN'), string(credentialsId: 'SORABOT_TOKEN', variable: 'SORABOT_TOKEN')]) {
       sonar_option = ""
-      if (CHANGE_ID != null)
-        sonar_option = "-Dsonar.github.pullRequest=${CHANGE_ID}"
+      if (scmVars.CHANGE_ID != null)
```
Comment (BulatSaif) [inline .jenkinsci-new/build.groovy]: CHANGE_ID  is not a scmVars 
```
scmVars = {GIT_BRANCH=PR-163, GIT_COMMIT=94c636211e477a3dd584bdeb1bb4a6212f51eb3e, GIT_LOCAL_BRANCH=PR-163, GIT_URL=https://github.com/hyperledger/iroha.git}
```

File: Jenkinsfile-new [hunk 2]
```diff
@@ -222,6 +224,21 @@ node ('master') {
...
+        testList = '()'
+        coverage = true
+        cppcheck = true
+        sonar = true
+        codestyle = true
+        useBTF = true
+        sanitize = true
+        packagePush=false
+        doxygen=false
+        break;
```
Comment (BulatSaif) [inline Jenkinsfile-new]: add `specialBranch=false` ,

File: .jenkinsci-new/build.groovy (hunk 1 above)
Comment (BulatSaif) [reply .jenkinsci-new/build.groovy]: `env.CHANGE_ID` should be the same

File: .jenkinsci-new/text-variables.groovy [hunk 3]
```diff
@@ -16,6 +16,7 @@ param_descriptions = """
   <strong>On open PR -</strong> Linux/gcc v5, MacOS/appleclang, Windows/msvc; Test: All; Coverage; Analysis: cppcheck, sonar, codestyle;<br />
   <strong>Commit in Open PR</strong> - Same as Branch commit<br />
   <strong>Before merge to trunk</strong> - Linux/gcc v5 v7, Linux/clang v6 v7, MacOS/appleclang, Windows/msvc; Test: ALL; Coverage; Analysis: cppcheck, sonar,codestyle; Build type: Debug when Release; useBTF=true<br />
+  <strong>Nightly build</strong> - Linux/gcc v5 v7, Linux/clang v6 v7, MacOS/appleclang, Windows/msvc; Test: ALL; Coverage; Analysis: cppcheck, sonar,codestyle; Build type: Debug when Release; useBTF=true; sanitize<br />
```
Comment (lebdron) [inline .jenkinsci-new/text-variables.groovy]: Not sure what does "Debug when Release" mean in "Build type".

**Pull request merged**